substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/elf/dwarf/DwarfInfoSectionImpl.java

```diff
@@ -794,7 +798,8 @@ private int writeMethodLocalDeclaration(DebugContext context, DebugLocalInfo par
         log(context, "  [0x%08x] method local declaration", pos);
         int abbrevCode;
         String paramName = paramInfo.name();
-        String paramTypeName = paramInfo.typeName();
+        TypeEntry paramType = lookupType(paramInfo.valueType());
+        String paramTypeName = paramType.getTypeName();
```

Comment:
Nitpick: Since `paramTypeName` is only used in the `log` calls I would inline it.

Reply:
Done. I also did the same with 2 occurrences of local var valueTypeName and one of local var returnTypeName that were only used in log statements.